### PR TITLE
Product Controller, Service, Repository.

### LINKED ISSUE #3: Allow basic product handling
Client should allow for CRUD operations for a basic item

A basic item should hold at least the following fields
- Title
- Description
- Category
- Price

### LINKED ISSUE #16: Allow CRUD requests for item
Allow CRUD requests that come from the rep client and from a user with appropriate role

### PR DESCRIPTION
**Pull request by Kaylesh**
Closes #16.

### CODE REVIEW

**Comment by markscamilleri** [comment]
Please don't link it to #3 as that's the frontend issue - which in fact would not be closed. I think you might mean to close #16?

File: backend/sprintone/src/main/java/com/project/omega/bean/Product.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    private String description;
+
+    @NotBlank
+    private String category;
+
+    @PositiveOrZero
+    private int price;
+
+    @OneToMany
+    @JoinColumn(name = "user_id", referencedColumnName = "id")
```
Comment (markscamilleri) [inline backend/sprintone/src/main/java/com/project/omega/bean/Product.java:30]: I know I'm not a reviewer, but what is the reason for a product to be linked to a user? Surely it makes more sense to link it to a supplier?

File: backend/sprintone/src/main/java/com/project/omega/controller/ProductController.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import com.project.omega.service.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "/product")
```
Comment (markscamilleri) [inline backend/sprintone/src/main/java/com/project/omega/controller/ProductController.java:18]: I'm also not understanding all these endpoints... Wouldn't it make sense to just have a `/product` (or `/products`)  and `/product/{id}` endpoint supporting `PUT`, `POST`, `GET` and `DELETE` methods for edit, new, get and delete respectively? The search parameters can be query parameters in the `GET` method.

Although this still works

File: backend/sprintone/src/main/java/com/project/omega/bean/Product.java (hunk 1 above)
Comment (Kaylesh) [reply backend/sprintone/src/main/java/com/project/omega/bean/Product.java:30]: This was prior to the introduction of a supplier entity into our model, and is scheduled for change. The thinking here at the time was to track which user had added the product via introducing their id as a foreign key at the time of product creation, purely as we did not (at the time) have the notion of a supplier entity.

File: backend/sprintone/src/main/java/com/project/omega/controller/ProductController.java (hunk 2 above)
Comment (Kaylesh) [reply backend/sprintone/src/main/java/com/project/omega/controller/ProductController.java:18]: Valid point, I will consider how best to consolidate the endpoints and update the rest docs accordingly.

**Pull request closed without merging**